pkg/machine-info/machine_info.go

```diff
@@ -0,0 +1,196 @@
+// Package machineinfo provides information about the machine.
+package machineinfo
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/shirou/gopsutil/v4/cpu"
+	"github.com/shirou/gopsutil/v4/mem"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	apiv1 "github.com/leptonai/gpud/api/v1"
+	"github.com/leptonai/gpud/pkg/asn"
+	pkgcontainerd "github.com/leptonai/gpud/pkg/containerd"
+	pkghost "github.com/leptonai/gpud/pkg/host"
+	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/netutil"
+	pkgnetutillatencyedge "github.com/leptonai/gpud/pkg/netutil/latency/edge"
+	nvidianvml "github.com/leptonai/gpud/pkg/nvidia-query/nvml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GetMachineInfo() (apiv1.MachineInfo, error) {
+	nvmlInstance, err := nvidianvml.New()
+	if err != nil {
+		return apiv1.MachineInfo{}, fmt.Errorf("failed to create nvml instance: %w", err)
+	}
+	defer func() {
+		if err := nvmlInstance.Shutdown(); err != nil {
+			log.Logger.Warnw("failed to shutdown nvml instance", "error", err)
+		}
```

Comment:
I am not sure if features that depend on nvml still work if we shutdown the nvml instance.

Reply:
Used to get driver + cuda version :)

changed it to debug